nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -331,7 +334,187 @@ internal static async Task<bool> DependenciesAreCoherentAsync(string repoRoot, s
         }
     }
 
-    internal static async Task<Dependency[]?> ResolveDependencyConflicts(string repoRoot, string projectPath, string targetFramework, Dependency[] packages, Logger logger)
+    internal static bool UseNewDependencySolver()
```

Comment:
nit: I think this method could just be simplified to `return Environment.GetEnvironmentVariable("UseNewNugetPackageResolver") == "true"`